plugins/catalog-backend/src/database/types.ts

```diff
@@ -81,6 +81,14 @@ export type ReplaceUnprocessedEntitiesOptions =
       type: 'delta';
     };
 
+export type RefreshKeyOptions = {
+  keys: { key: String; entity: Entity }[];
```

Comment:
Could be, however in this case it would mean we'd need to emit the entityRefs from the processors, so the entityRef construction would happen inside all of the processors. With the current implementation they just emit an entity, and in the processing engine we can construct the refs based on the emitted entity. 

I'm fine with both of them, maybe leaning towards the emission of entityRefs just for the sake of reducing the data thatis sent :D